packages/ai/core/tool/mcp/mcp-sse-transport.ts

```diff
@@ -147,6 +150,9 @@ export class SSEClientTransport implements MCPTransport {
     try {
       const headers = new Headers();
       headers.set('Content-Type', 'application/json');
+      Object.entries(this.headers ?? {}).forEach(([key, value]) => {
```

Comment:
nit: would be cleaner to do something like:

`const headers = new Headers({ ...this.headers })` and then set the content type